### ISSUE TITLE
Use pre trained word2vec vector as input

### ISSUE THREAD

**Issue by BrikerMan** [chsasank/ATIS.keras#6]
I am trying to use word2vec vector as input.

this is the original input
```
 1     ['我', '想', '看', '电影']
 2     ['有', '没有', '美剧']
 ...
 9999  ['有', '没有', '美剧']
```

replace words with the vector
```
 1     [[word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200]]
 2     [[word2vec size=200], [word2vec size=200], [word2vec size=200]]
 ...
 9999  [[word2vec size=200], [word2vec size=200], [word2vec size=200]]
```

pan the vectors with np.zeros.
```
 1     [[word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200]]
 2     [[word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200]]
 ....
 9999  [[word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200], [word2vec size=200]]
```

Then use the result as input. Is this doable? and how should I do that? Thanks a lot.

**Comment by chsasank** [comment]
But the code already has a word embeddings layer.
I think you can init it with your word2vec embeddings and freeze that layer to produce the desired effect.

**Comment by BrikerMan** [comment]
I know your code has own embeddings layer. But I need to embedding word using the train data and other data for accuracy. Not all the data I use for embedding will use to slot filling. So I have to do this separately.

**Comment by chsasank** [comment]
Yep, got it. Two options: 
1. remove the Embedding layer and pass the embedded inputs instead of indexes
2. Initialise the weights on embedding using `embeddings_initializer` argument and [freeze it](https://keras.io/getting-started/faq/#how-can-i-freeze-keras-layers)

**Comment by BrikerMan** [comment]
I will try both ways. Thanks very much. I am new to ML. And have lots of
problems. Is that ok to contact you if I have any new questions?
Sasank Chilamkurthy <notifications@github.com>于2017年7月21日 周五下午7:51写道：

**Comment by BrikerMan** [comment]
@chsasank I have successfully initialized the weights on embedding with a pre-trained word2Vec model, it works fine. Thanks for your suggestion~

**Comment by alepacheco** [comment]
Hi @BrikerMan How did you managed to use a pretrained word2Vec model?

**Comment by BrikerMan** [comment]
@alepacheco https://eliyar.biz/using-pre-trained-gensim-word2vector-in-a-keras-model-and-visualizing/ here

**Comment by alepacheco** [comment]
@BrikerMan I read your post but there you don't use the ATIS dataset, I'm trying to find a way to use the two models together to slot-fill words that are not on the ATIS dataset (ex: more cities) and classifly them in departure city, destination, etc... 

Thanks for your help :smiley:

**Comment by BrikerMan** [comment]
@alepacheco I am understanding that you want to classify data and slot just like ATIS with more city name, right?

**Comment by alepacheco** [comment]
@BrikerMan Yes, because there are not many words on the ATIS dataset I want to use Word2vec to encode the words and train this network and be able to slot unseen words. I tried changing the embedding layer to use word2vec and run the model with arrays of Word2vecVectors but the results have the following shape (18, 300, 127). 18 been the sentence length,  300 been the dimensions of the words vectors and 127 been the class types. So I'm stuck trying to understand this

**Comment by BrikerMan** [comment]
I understand your problem. You need to load a well trained word2vec model to the embedding layer. If is it ok to share your code with me, maybe I can help. Content my email eliyar917#gmail.com.
